src/include/uapi/ipc/control.h

```diff
@@ -165,6 +165,71 @@ struct sof_ipc_ctrl_data {
 	};
 } __attribute__((packed));
 
+/**
+ * Error information reported back for APP_CONFIG_PARAM
+ * (aka VENDOR_CONFIG_PARAM) inside the struct sof_ipc_large_data.
+ */
+struct sof_ipc_large_data_status {
+	uint32_t failed_param_index;	/**< index of the failed parameter */
+	uint32_t app_error_code;	/**< app error code */
+} __attribute__((packed));
+
+/**
+ * Application parameter transferred inside struct sof_ipc_large_data.
+ *
+ * <i>priv_param_id</i> bits are assigned as follows:
+ * - [31-8] : parameter instance,
+ * - [7-0]  : parameter type
+ *
+ * In case of GET operation, specifies maximum parameter size accepted
+ * by the host application. When multiple parameters are tunneled inside
+ * a single request/response, the total size of parameters must fit the
+ * single IPC message payload.
+ */
+struct sof_ipc_large_data_app_param {
+	uint32_t param_id;	/**< parameter id */
+	uint32_t size;		/**< size in bytes */
+	uint32_t data[0];	/**< data */
+} __attribute__((packed));
```

Comment:
I don't get the GET...

it looks like this is used as input and output?
How would multiple parameters be tunneled? In which order?

Reply:
If you decide to handle multiple parameters in one get/set driver api call, then you may send them down in a one or more requests depending on how many would fit into the IPC message buffer. The order defined by the driver api should be preserved I guess.

Reply:
Should be using the ABI header for this (already defined).

Reply:
I also don't get how you encode multiple parameters with the structure above?